fem/pgridfunc.cpp

```diff
@@ -130,7 +131,15 @@ void ParGridFunction::MakeRef(ParFiniteElementSpace *f, Vector &v, int v_offset)
 
 void ParGridFunction::Distribute(const Vector *tv)
 {
-   pfes->GetProlongationMatrix()->Mult(*tv, *this);
+   const Operator *prolong = pfes->GetProlongationMatrix();
+   if (!IsIdentityProlongation(prolong))
+   {
+      prolong->Mult(*tv, *this);
+   }
+   else
+   {
+      *this = *tv;
+   }
```

Comment:
This can now be simplified to just `prolong->Mult(*tv, *this);` -- the pointer cannot be `NULL` now.